extensions-core/multi-stage-query/src/test/java/org/apache/druid/msq/exec/MSQComplexGroupByTest.java

```diff
@@ -416,4 +432,91 @@ public void testSortingOnNestedData(String contextName, Map<String, Object> cont
                      ))
                      .verifyResults();
   }
+
+  @MethodSource("data")
+  @ParameterizedTest(name = "{index}:with context {0}")
+  public void testCountDistinctOnNestedData(String contextName, Map<String, Object> context)
```

Comment:
Should probably add to the test name that this is an exact distinct since the approximate aggs don't work with complex types yet afaik.

it might be interesting to also add a test where every row is not unique, such as the `cObj` column in this same file, though i suppose that is the other extreme...